Automattic/wp-calypso
# Domain step test: Add the free domain for a year explainer textbox

#### Changes proposed in this Pull Request

This PR builds on top of the domain-copy A/B test introduced in https://github.com/Automattic/wp-calypso/pull/37546. Check the `What next?` section in that PR on how to review this and other PRs as part of this A/B test.

* Add the free domain for a year explainer textbox below the domain input box. The textbox is shown only when there are domain search results displayed.
* The CTA on the textbox will take you to the plans page with the free site option hidden(as discussed in pbm1bU-2G#comment-67-p2)
* For detailed context, check p8Eqe3-Uq-p2.

<img width="1375" alt="Screenshot 2019-11-20 at 8 24 31 PM" src="https://user-images.githubusercontent.com/1269602/69250202-41512d00-0bd5-11ea-950f-bc678ca2b0ca.png">


#### Testing instructions



* Go to /start and put yourself in the variant variantShowUpdates of the domainStepCopyUpdates A/B test.
* The free-domain explainer textbox should be displayed with the same format and spacing alignments as shown in the screenshot above.
* Click the `Review plan options ...` link and verify that the plan page on the next step _does not show_ the free site option

<img width="680" alt="Screenshot 2019-11-15 at 8 37 18 PM" src="https://user-images.githubusercontent.com/1269602/68954356-e04ae300-07e9-11ea-8055-a3897dccb7d0.png">

* Complete signup after clicking `Review plan options ...` and verify that a new site is created for you.


**Mobile View**

* Verify alignment and copy checks out in mobile views.

**Check in other domain pages**

* Login to any account, and after assigning yourself to the variant, check the launch flow for privates sites and the Add domain page. In both places, the domain step copy changes should not be seen(as we are showing the copy only during signup).

**Verify that the A/B test is not automatically assigned**
* Go to /start and without manually assigning yourself to an A/B test, just complete the signup flow. 
* In your Browser console, type `localstorage.ABTests;`. Verify that the `domainStepCopyUpdates ` test is not present in the object. This is because we do not yet want any user getting assigned to the test till all the pieces are deployed.
